### PR TITLE
Update index-ml.md

### PR DESCRIPTION
**Pull request by khanteymoori**
After Anup's modifications,
new categories are added: dimension reduction, model building, and visualization
new tools are added related to dimension reduction and visualization
and the tools are re-categorized

### CODE REVIEW

**Comment by bgruening** [comment]
@anuprulez can you also look at this?

@khanteymoori thanks.

File: index-ml.md [hunk 1]
```diff
@@ -99,32 +90,59 @@ Tool | Description | Reference
...
 
-Comparing, validating and choosing parameters and models.
+Building general machine learning models.
+
+Tool | Description | Reference
+--- | --- | ---
+{% include tool.html id="Estimator Attributes" %} | Estimator attributes get important attributes from an estimator or scikit object | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Stacking Ensemble Models" %} | Stacking Ensembles builds stacking, voting ensemble models with numerous base options | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Search CV" %} | Hyperparameter Search performs hyperparameter optimization using various SearchCVs  | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Build Pipeline" %} | Pipeline Builder an all-in-one platform to build pipeline, single estimator, preprocessor and custom wrappers | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
```
Comment (anuprulez) [inline index-ml.md]: `Pipeline Builder as an all-in-one ...`

File: index-ml.md [hunk 2]
```diff
@@ -99,32 +90,59 @@ Tool | Description | Reference
...
 
 Tool | Description | Reference
 --- | --- | ---
 {% include tool.html id="Model validation" %} | Model Validation includes cross_validate, cross_val_predict, learning_curve, and more | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
-{% include tool.html id="CLF Metrics" %} | Calculate metrics for classification performance  | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
 {% include tool.html id="Pairwise Metrics" %} | Evaluate pairwise distances or compute affinity or kernel for sets of samples | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
-{% include tool.html id="Regression metrics" %} | Calculate metrics for regression performance | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Train/Test evaluation" %} | Train, Test and Evaluation fit a model using part of dataset and evaluate using the rest | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
```
Comment (anuprulez) [inline index-ml.md]: `Train, Test and Evaluation to fit a model ... `

File: index-ml.md [hunk 3]
```diff
@@ -99,32 +90,59 @@ Tool | Description | Reference
...
 
-## Model selection
+## Model building
 
-Comparing, validating and choosing parameters and models.
+Building general machine learning models.
+
+Tool | Description | Reference
+--- | --- | ---
+{% include tool.html id="Estimator Attributes" %} | Estimator attributes get important attributes from an estimator or scikit object | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
```
Comment (anuprulez) [inline index-ml.md]: `Estimator attributes to get all attributes ... `

File: index-ml.md [hunk 4]
```diff
@@ -99,32 +90,59 @@ Tool | Description | Reference
...
+{% include tool.html id="Stacking Ensemble Models" %} | Stacking Ensembles builds stacking, voting ensemble models with numerous base options | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
```
Comment (anuprulez) [inline index-ml.md]: `Stacking Ensembles to build stacking ... `

File: index-ml.md (hunk 2 above)
Comment (anuprulez) [reply index-ml.md]: Similar comments for all other sentences containing names of tools and their descriptions

File: index-ml.md [hunk 5]
```diff
@@ -134,16 +152,28 @@ Tool | Description | Reference
...
+
 ## Utilities
 
-General data manipulation tools.
+General data and table manipulation tools.
 
 Tool | Description | Reference
 --- | --- | ---
 {% include tool.html id="Table compute" %} | The power of the pandas data library for manipulating and computing expressions upon tabular data and matrices. |
 {% include tool.html id="datamash operations" %} | datamash operations on tabular data |
```
Comment (anuprulez) [inline index-ml.md]: `Datamash operations on tabular data `

File: index-ml.md [hunk 6]
```diff
@@ -99,32 +90,59 @@ Tool | Description | Reference
...
+{% include tool.html id="kpca1" %} | Kernel Principal Component Analysis|
+{% include tool.html id="cca1" %} | Canonical Correlation Analysis|
+{% include tool.html id="kcca1" %} | Kernel Canonical Correlation Analysis|
+{: .table.table-striped .tooltable}
 
-## Model selection
+## Model building
 
-Comparing, validating and choosing parameters and models.
+Building general machine learning models.
```
Comment (anuprulez) [inline index-ml.md]: `Building machine learning models`

File: index-ml.md (hunk 5 above)
Comment (khanteymoori) [reply index-ml.md]: all changes are done

**Pull request merged**